dspace-api/src/test/java/org/dspace/content/BundleTest.java

```diff
@@ -513,6 +513,41 @@ public void testRemoveBitstreamAuth() throws SQLException, AuthorizeException, I
     }
 
 
+    /**
+     * Test removeBitstream method and also the unsetPrimaryBitstreamID method, of class Bundle.
+     */
+    @Test
+    public void testRemoveBitstreamAuthAndUnsetPrimaryBitstreamID()
+            throws IOException, SQLException, AuthorizeException {
+        // Allow Item WRITE permissions
+        doNothing().when(authorizeServiceSpy).authorizeAction(context, item, Constants.WRITE);
+        // Allow Bundle ADD permissions
+        doNothing().when(authorizeServiceSpy).authorizeAction(context, b, Constants.ADD);
+        // Allow Bundle REMOVE permissions
+        doNothing().when(authorizeServiceSpy).authorizeAction(context, b, Constants.REMOVE);
+        // Allow Bitstream WRITE permissions
+        doNothing().when(authorizeServiceSpy)
+                   .authorizeAction(any(Context.class), any(Bitstream.class), eq(Constants.WRITE));
+        // Allow Bitstream DELETE permissions
+        doNothing().when(authorizeServiceSpy)
+                   .authorizeAction(any(Context.class), any(Bitstream.class), eq(Constants.DELETE));
+
+
+        context.turnOffAuthorisationSystem();
+        //set a value different than default
+        File f = new File(testProps.get("test.bitstream").toString());
+        Bitstream bs = bitstreamService.create(context, new FileInputStream(f));
+        bundleService.addBitstream(context, b, bs);
+        b.setPrimaryBitstreamID(bs);
+        context.restoreAuthSystemState();
+
+        assertThat("testRemoveBitstreamAuthAndUnsetPrimaryBitstreamID 0", b.getPrimaryBitstream(), equalTo(bs));
+        //remove bitstream
+        bundleService.removeBitstream(context, b, bs);
```

Comment:
I think it would be good to do a similar test but directly testing  `bitstreamService.delete(context, bs)` instead of `bundleService.removeBitstream(context, b, bs)`

Reply:
Please check commit:
https://github.com/DSpace/DSpace/pull/9106/commits/e6d108a94e41e58d6d701f3ef0429fda438e6555